src/pages/RepoPage/BundlesTab/BundleContent/BundleSummary/hooks/useBundleBranchSelector.ts

```diff
@@ -0,0 +1,59 @@
+import { useParams } from 'react-router-dom'
+
+import { Branch, useBranch } from 'services/branches'
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+  branch?: string
+  ref?: string
+}
+
+const getDecodedBranch = (branch?: string) =>
```

Comment:
nit: if branch and ref don't exist, should we be explicit about the "falsey" return value we want from the ternary? Like '' or similar

Reply:
We can't use `""`, but we can return `undefined` instead

Reply:
'' is what would have been returned from the conditional if branch was ''. Which is what spawned the question in the first place :) 

If that's what we want then I'm aligned

![Screenshot 2024-02-20 at 2 07 12 PM](https://github.com/codecov/gazebo/assets/159853603/2bf08420-ea23-4eed-8d4b-c05d252574b9)

Reply:
So one small thing that I did also change is to swap to using the `??` nullish coalescing operator which only fails the left hand side if `null` or `undefined` as it is a bit safer to use over the typical `||` logical OR.

Reply:
Yep yep totally makes sense!